core/array.cpp

```diff
@@ -35,8 +35,8 @@
 #include "variant.h"
 #include "vector.h"
 
-struct ArrayPrivate {
-
+class ArrayPrivate {
```

Comment:
Why? In C++ a Class with only public members is perfectly equivalent to a struct in every way.

Reply:
Clang complains otherwise.